server/jwt_test.go

```diff
@@ -1691,6 +1691,166 @@ func TestAccountURLResolverNoFetchOnReload(t *testing.T) {
+		}
+	}))
+	defer ts.Close()
+	// Create seed server A
+	confA := createConfFile(t, []byte(fmt.Sprintf(`
+		listen: -1
+		operator: %s
+		resolver: URL("%s/A/")
+		cluster {
+			name: clust
+			listen: -1
+		}
+    `, ojwt, ts.URL)))
+	defer os.Remove(confA)
+	//optsA := LoadConfig(confA)
+	//optsA.NoLog = false
+	//sA := RunServer(optsA)
+	sA, _ := RunServerWithConfig(confA)
+	defer sA.Shutdown()
+	// Create Server B
+	confB := createConfFile(t, []byte(fmt.Sprintf(`
+		listen: -1
+		operator: %s
+		resolver: URL("%s/B/")
+		cluster {
+			name: clust
+			listen: -1 
+			routes [
+				nats-route://localhost:%d
+			]
+		}
+    `, ojwt, ts.URL, sA.opts.Cluster.Port)))
+	defer os.Remove(confB)
+	optsB := LoadConfig(confB)
+	//optsB.NoLog = false
+	//optsB.Debug = true // only this will show a trace hinting at the issue
+	// Unknown account "<acc>" for subject "subscription.loss.test"
+	sB := RunServer(optsB)
+	//sB, _ := RunServerWithConfig(confB)
+	defer sB.Shutdown()
+	// startup cluster
+	checkClusterFormed(t, sA, sB)
+	// Both server observed one fetch on startup
+	if path := <-chanA; path != "/A/" {
+		t.Fatal("Expect one fetch on A", path)
+	}
+	if path := <-chanB; path != "/B/" {
+		t.Fatal("Expect one fetch on B", path)
+	}
+	// Create first client, directly connects to A
+	urlA := fmt.Sprintf("nats://%s:%d", sA.opts.Host, sA.opts.Port)
+	ncA, err := nats.Connect(urlA, nats.UserCredentials(creds),
+		nats.DisconnectErrHandler(func(_ *nats.Conn, err error) {
+			if err != nil {
+				t.Fatal("error not expected in this test", err)
+			}
+		}),
+		nats.ErrorHandler(func(_ *nats.Conn, _ *nats.Subscription, err error) {
+			t.Fatal("error not expected in this test", err)
+		}),
+	)
+	if err != nil {
+		t.Fatalf("Expected to connect, got %v %s", err, urlA)
+	}
+	defer ncA.Close()
+	// create a test subscription
+	subA, err := ncA.SubscribeSync("subscription.loss.test")
+	if err != nil {
+		t.Fatalf("Expected no error during subscribe: %v", err)
+	}
+	defer subA.Unsubscribe()
+	// Both server observed one fetch for the connecting client
+	if path := <-chanA; path != "/A/" {
+		t.Fatal("Expect one fetch on A")
+	}
+	if path := <-chanB; path != "/B/" {
+		t.Fatal("Expect one fetch on B")
+	}
+	// create second client, directly connect to B
+	urlB := fmt.Sprintf("nats://%s:%d", sB.opts.Host, sB.opts.Port)
+	ncB, err := nats.Connect(urlB, nats.UserCredentials(creds))
+	if err != nil {
+		t.Fatalf("Expected to connect, got %v %s", err, urlB)
+	}
+	defer ncB.Close()
+	// Because the first request was a failure, a second request is observer as a result of connect to B
+	// This needs to be commented out when there is a fix
+	for i := int32(0); i < errCnt; i++ {
+		if path := <-chanB; path != "/B/" {
+			t.Fatal("Expect one fetch on B")
+		}
+	}
+	// the route subscription was lost due to the failed fetch
+	// Noe we test if some recover mechanism is in play
+	if err := ncB.Publish("subscription.loss.test", []byte("msg")); err != nil {
```

Comment:
If we do close the connection, I would call `checkClusterFormed()` here again prior to publish. Otherwise you may get a flapper (even with the fix). You technically would also check the there is the subscription interest on "B" (use existing helper: `func checkSubInterest(t *testing.T, s *Server, accName, subject string, timeout time.Duration)`).